### PR TITLE
Drop modals from V2: Create Org page

### PR DESCRIPTION
**Pull request by TuvalSimha**
@kamilkisiela 

## Things i did here:

- Split Create org modal:
1. `CreateOrganizationForm`: handle all graphql mutation, functions, etc
2. `CreateOrganizationFormContent`: handle the UI only, and get the form, functions as props
- Added `CreateOrganizationFormContent` to Storybook to display the modal and form behaviour. [LINK](https://b227ef0b.hive-storybook.pages.dev/?path=/story/modals-create-organization-modal--default)
- Spinner component: part of the form modal, moved from V2 component to UI components.
- Support `__ENV` for Storybook in `preview.js`

### CODE REVIEW

File: packages/web/app/src/components/ui/modals/create-organization.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { useForm } from 'react-hook-form';
+import { useForm, UseFormReturn } from 'react-hook-form';
```
Comment (kamilkisiela) [inline packages/web/app/src/components/ui/modals/create-organization.tsx:1]: Let's maybe move this file closer to the file it's used in and move away from this `/ui/modals/*.tsx` pattern. These modals are not really used in more than one place. 
It's a good opportunity to clean things up.

File: packages/web/app/src/components/ui/modals/create-organization.tsx [hunk 2]
```diff
@@ -104,50 +104,64 @@ export const CreateOrganizationForm = () => {
     }
   }
 
+  return <CreateOrganizationFormContent form={form} onSubmit={() => onSubmit(form.getValues())} />;
+};
+
+type CreateOrganizationFormContentProps = {
+  form: UseFormReturn<z.infer<typeof formSchema>>;
+  onSubmit: () => void;
```
Comment (kamilkisiela) [inline packages/web/app/src/components/ui/modals/create-organization.tsx:112]:
```suggestion
  onSubmit: (values: z.infer<typeof formSchema>) => void | Promise<void>;
```

File: packages/web/app/src/components/ui/modals/create-organization.tsx [hunk 3]
```diff
@@ -104,50 +104,64 @@ export const CreateOrganizationForm = () => {
     }
   }
 
+  return <CreateOrganizationFormContent form={form} onSubmit={() => onSubmit(form.getValues())} />;
```
Comment (kamilkisiela) [inline packages/web/app/src/components/ui/modals/create-organization.tsx:107]:
```suggestion
  return <CreateOrganizationFormContent form={form} onSubmit={onSubmit} />;
```

Comment (kamilkisiela) [reply packages/web/app/src/components/ui/modals/create-organization.tsx:107]: the `form` may or may not be in sync with submitted values, we assume it is, but the callback (called by `form.handleSubmit`) might do weird things.

File: packages/web/app/src/pages/organization-new.tsx [hunk 4]
```diff
@@ -3,34 +3,31 @@ import { LogOutIcon } from 'lucide-react';
...
 import { Link, useRouter } from '@tanstack/react-router';
 
 export function NewOrgPage(): ReactElement {
   const router = useRouter();
+
+  const handleOnClick = (): void => {
+    void router.navigate({
+      to: '/logout',
+    });
+  };
```
Comment (kamilkisiela) [inline packages/web/app/src/pages/organization-new.tsx:17]: It's fine to leave it as is, but it does not make sense to turn those small callbacks into dedicated functions, maybe if the name was different (like "onLogoutClick" or something), but still

**Review by kamilkisiela — APPROVED** [review]
Once you move the component, feel free to merge the PR

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**